opm/autodiff/GeoProps.hpp

```diff
@@ -372,11 +372,27 @@ namespace Opm
                 double cn = 0.0;
                 double sgn = 2.0 * (faceCells(faceIdx, 0) == cellIdx) - 1;
                 const int dim = Opm::UgGridHelpers::dimensions(grid);
+
+                const double* faceNormal = Opm::UgGridHelpers::faceNormal(grid, faceIdx);
+#if HAVE_DUNE_CORNERPOINT
+                assert( dim <= 3 );
+                Dune::FieldVector< double, 3 > scaledFaceNormal;
+                for (int indx = 0; indx < dim; ++indx) {
+                    scaledFaceNormal[ indx ] = faceNormal[ indx ];
+                }
+                // compute unit normal incase the normal is already scaled
+                scaledFaceNormal /= scaledFaceNormal.two_norm();
```

Comment:
Will this be correct with dim == 2? I think it is correct if FieldVector initialises to zeros by default, but I think it leaves the third element uninitialised?

If so, I am ok with changing just the assertion to require dim == 3.